### PR TITLE
Feat/sapphire updates

### PR DESCRIPTION
**Pull request by grvgoel81**
Feat/sapphire updates

### CODE REVIEW

File: Assets/Plugins/Web3AuthSDK/Types/WhiteLabelData.cs [hunk 1]
```diff
@@ -1,10 +1,12 @@
 ﻿using System.Collections.Generic;
 
 public class WhiteLabelData { 
-    public string name { get; set; }
+    public string appName { get; set; }
     public string logoLight { get; set; }
     public string logoDark { get; set; }
     public string defaultLanguage { get; set; } = "en";
-    public bool dark { get; set; } = false;
+    public string mode { get; set; } = "light";
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Types/WhiteLabelData.cs:8]: should be enum

Comment (chaitanyapotti) [reply Assets/Plugins/Web3AuthSDK/Types/WhiteLabelData.cs:8]: all whitelabel fields are optional

File: Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs [hunk 2]
```diff
@@ -4,19 +4,25 @@
...
-            if (network == Web3Auth.Network.TESTNET)
-                return "https://dev-sdk.openlogin.com";
-            else
-                return "https://sdk.openlogin.com";
+            if (buildEnv == Web3Auth.BuildEnv.STAGING)
+                return "https://staging-auth.web3auth.io/{openLoginVersion}";
+            else if (buildEnv == Web3Auth.BuildEnv.TESTING)
+                return "https://develop-auth.web3auth.io";
+            else 
+                return "https://auth.web3auth.io/{openLoginVersion}";
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs:17]: will this work?

File: Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs [hunk 3]
```diff
@@ -4,19 +4,25 @@
...
         }
         set { }
     }
+    public const string openLoginVersion = "v5";
 
     public WhiteLabelData whiteLabel { get; set; }
     public Dictionary<string, LoginConfigItem> loginConfig { get; set; }
     public bool? useCoreKitKey { get; set; } = false;
     public Web3Auth.ChainNamespace? chainNamespace { get; set; } = Web3Auth.ChainNamespace.EIP155;
+    public MfaSettings mfaSettings { get; set; } = null;
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs:27]: optionals for whitelabel, loginconfig, mfasettings etc.

Comment (chaitanyapotti) [reply Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs:27]: default for buildenv is production

File: Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs (hunk 2 above)
Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs:17]: Yes, tested by passing buildEnv as production

File: Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs (hunk 3 above)
Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Types/Web3AuthOptions.cs:27]: Resolved

File: Assets/Plugins/Web3AuthSDK/Types/WhiteLabelData.cs (hunk 1 above)
Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Types/WhiteLabelData.cs]: > all whitelabel fields are optional

Resolved

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 4]
```diff
@@ -69,29 +85,35 @@ public void Awake()
...
 
         if (this.web3AuthOptions.loginConfig != null)
-            this.initParams["loginConfig"] = JsonConvert.SerializeObject(this.web3AuthOptions.loginConfig);
+            this.initParams["loginConfig"] = JsonConvert.SerializeObject(this.web3AuthOptions.loginConfig, settings);
 
         if (this.web3AuthOptions.clientId != null)
             this.initParams["clientId"] = this.web3AuthOptions.clientId;
 
-        this.initParams["network"] = this.web3AuthOptions.network.ToString().ToLower();
+        if (this.web3AuthOptions.clientId != null)
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Web3Auth.cs:113]: check this.. should be `this.web3AuthOptions.buildEnv`

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 5]
```diff
@@ -69,29 +85,35 @@ public void Awake()
 //            this.setResultUrl(new Uri($"http://localhost#{code}"));
 //        } 
 #endif
-
-        authorizeSession();
+        authorizeSession("");
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Web3Auth.cs:88]: Add all other params to Awake function as well. initParams should be populated correctly

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 6]
```diff
@@ -427,6 +462,54 @@ private void sessionTimeOutAPI()
...
+        var encryptedData = aes256cbc.encrypt(System.Text.Encoding.UTF8.GetBytes(data));
+        var mac = aes256cbc.getMac(encryptedData);
+        var encryptedMetadata = new ShareMetadata()
+        {
+            iv = KeyStoreManagerUtils.convertByteToHexadecimal(ivKey),
+            ephemPublicKey = ephemKey,
+            ciphertext = KeyStoreManagerUtils.convertByteToHexadecimal(encryptedData),
+            mac = KeyStoreManagerUtils.convertByteToHexadecimal(mac)
+        };
+        var jsonData = JsonConvert.SerializeObject(encryptedMetadata);
```
Comment (chaitanyapotti) [inline Assets/Plugins/Web3AuthSDK/Web3Auth.cs:484]: might need to add a delay here so that the api call finishes.. before responding..

Comment (chaitanyapotti) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs:484]: like 200 ms

Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs:484]: Why delay above? Code is executing serially and jsonData is being passed in API call.

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs (hunk 5 above)
Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs:88]: resolved.

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs (hunk 4 above)
Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs]: Resolved

File: Assets/Plugins/Web3AuthSDK/Web3Auth.cs (hunk 6 above)
Comment (chaitanyapotti) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs:484]: you're not waiting for coroutine to finish. `yield` is the keyword.
https://docs.unity3d.com/ScriptReference/MonoBehaviour.StartCoroutine.html

Comment (grvgoel81) [reply Assets/Plugins/Web3AuthSDK/Web3Auth.cs:485]: resolved.

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**